modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java

```diff
@@ -173,15 +173,17 @@ void updateDatabases() throws IOException {
                 final String id = entry.getKey();
                 DatabaseConfiguration database = entry.getValue().database();
                 if (existingDatabaseNames.contains(database.name() + ".mmdb") == false) {
-                    logger.debug("A new database appeared [{}]", database.name());
+                    if (database.provider() instanceof DatabaseConfiguration.Maxmind maxmind) {
+                        logger.debug("A new database appeared [{}]", database.name());
+                        final String accountId = maxmind.accountId();
+                        try (HttpClient.PasswordAuthenticationHolder holder = credentialsBuilder.apply(accountId)) {
+                            if (holder == null) {
```

Comment:
We're getting quite a few layers deep here. Is it worth extracting some of these blocks to methods for readability?

Reply:
You caught us in the middle of a few changes ::) -- @joegallo took care of this in another PR that I just merged into this one.